mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -104,46 +106,66 @@ void AffixTuningParameters::runOnFunction() {
   func.walk([&](miopen::GridwiseGemmOp op) {
     OpBuilder b(op.getContext());
 
-    // TBD. Compute tuning parameters from actual logic.
-    op.setAttr("block_size", b.getI32IntegerAttr(256));
+    ConvolutionContext convContext = populateConvContext(op);
+    InitParamsNonXDL validParams{0, 0, 0, 0, 0, 0};
+    GemmSize gemmSize;
+    DerivedParams gemmADerivedParam;
+    DerivedParams gemmBDerivedParam;
+    int64_t gemmCDstPerWrite;
+    int64_t gridSize;
+
+    PopulateParams populateParams;
+    LogicalResult status = populateParams.paramsFromCtx(
+        convContext, validParams, gemmSize, gemmADerivedParam,
+        gemmBDerivedParam, gemmCDstPerWrite, gridSize);
+    if (failed(status)) {
+      signalPassFailure();
+    }
 
-    op.setAttr("m_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("n_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("k_per_block", b.getI32IntegerAttr(8));
+    // TODO: Override the block size and grid size before populating params.
+    // The way it is implemented now cannot guarantee cohereancy of tuning
+    // parameters
+    int64_t grid_size = gridSize;
+    int64_t block_size = validParams.blockSize;
+    if (launchDimCallback) {
+      launchDimCallback(block_size, grid_size);
```

Comment:
perhaps just `launchDimCallback(validParams.blockSize, gridSize)`?

Reply:
Yes, I can do that to simplify the call back assignment, but the current implementation is not right. Keeping them separate help me remember they carry different meanings. 

My immediate next PR plans to:

- Make this callback happens before the parameter populate
- Update the heuristic loop such that it will only respond to blockSize and gridSize similar to the passed in overrides.

Only after I have done the above can I guarantee both grid size and block size are set to correct coherent values.